nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```
@@ -0,0 +1,310 @@
+    mask_pose.y = out.point.y;
+  } else {
+    // Filter mask and current layer are in the same frame:
+    // Just use pose coordinates
+    mask_pose = pose;
+  }
+
+  return true;
+}
+
+bool BinaryFilter::worldToMask(double wx, double wy, unsigned int & mx, unsigned int & my) const
+{
+  double origin_x = filter_mask_->info.origin.position.x;
+  double origin_y = filter_mask_->info.origin.position.y;
+  double resolution = filter_mask_->info.resolution;
+  unsigned int size_x = filter_mask_->info.width;
+  unsigned int size_y = filter_mask_->info.height;
+
+  if (wx < origin_x || wy < origin_y) {
+    return false;
+  }
+
+  mx = std::round((wx - origin_x) / resolution);
+  my = std::round((wy - origin_y) / resolution);
+  if (mx >= size_x || my >= size_y) {
+    return false;
+  }
+
+  return true;
+}
+
+inline int8_t BinaryFilter::getMaskData(
+  const unsigned int mx, const unsigned int my) const
+{
+  return filter_mask_->data[my * filter_mask_->info.width + mx];
+}
+
+void BinaryFilter::process(
+  nav2_costmap_2d::Costmap2D & /*master_grid*/,
+  int /*min_i*/, int /*min_j*/, int /*max_i*/, int /*max_j*/,
+  const geometry_msgs::msg::Pose2D & pose)
+{
+  std::lock_guard<CostmapFilter::mutex_t> guard(*getMutex());
+
+  if (!filter_mask_) {
+    // Show warning message every 2 seconds to not litter an output
+    RCLCPP_WARN_THROTTLE(
+      logger_, *(clock_), 2000,
+      "BinaryFilter: Filter mask was not received");
+    return;
+  }
+
+  geometry_msgs::msg::Pose2D mask_pose;  // robot coordinates in mask frame
+
+  // Transforming robot pose from current layer frame to mask frame
+  if (!transformPose(pose, mask_pose)) {
+    return;
+  }
+
+  // Converting mask_pose robot position to filter_mask_ indexes (mask_robot_i, mask_robot_j)
+  unsigned int mask_robot_i, mask_robot_j;
+  if (!worldToMask(mask_pose.x, mask_pose.y, mask_robot_i, mask_robot_j)) {
+    // Robot went out of mask range. Set "false" state by-default
+    changeState(false);
```

Use default value -- also log a warning here that we're outside of the mask.

---

Done in https://github.com/ros-planning/navigation2/pull/3228/commits/056ed5e09f0092855591bed2223511fc8d0cac71